packages/core/src/modules/credentials/repository/CredentialRecord.ts

```diff
@@ -97,6 +95,22 @@ export class CredentialRecord extends BaseRecord<DefaultCredentialTags, CustomCr
       this.credentialAttributes = props.credentialAttributes
       this.autoAcceptCredential = props.autoAcceptCredential
       this.linkedAttachments = props.linkedAttachments
+
+      if (props.metadata) {
+        if (props.metadata.requestMetadata) {
+          this.metadata.set('requestMetadata', props.metadata.requestMetadata)
+        }
+        if (props.metadata.schemaId) {
+          this.metadata.add('indyCredentialMetadata', {
+            schemaId: props.metadata.schemaId,
+          })
+        }
+        if (props.metadata.credentialDefinitionId) {
+          this.metadata.add('indyCredentialMetadata', {
+            credentialDefinitionId: props.metadata.credentialDefinitionId,
+          })
+        }
+      }
```

Comment:
It may be wise to increment our version in some manner to indicate a breaking change here, since this could cause fairly significant issues for existing systems? Is there harm in utilizing @blu3beri's original mechanism until a later date?

Reply:
I think @blu3beri's approach wouldn't work out of the box. The constructor is not called with properties when a record is retrieved from storage. We could add a custom transformer, but I'd rather only take these things into account between stable releases.

Maybe we can look at writing transform scripts. I think we should be very cautious with adding these if statements to the codebase. I don't want to be stuck with this piece of code for eternity. We can never assume when all records are transformed. We're going to have a lot more breaking changes going forward that I'll rather solve by writing update scripts. I'd day that is a reasonable for pre-1.0 releases

Maybe we can keep it in here till 0.1.0. Then for 0.1.0 we provide scripts, and for 0.2.0 we again provide a script for the transformation.

Thoughts?

Reply:
Maybe I am misunderstanding your point, but right now we set this data in the constructor already and the `credentialService`, line 612 and 627, relay on these fields. It will even error out when we do not have these fields in the metadata. 

I am okay with leaving this out, but how do you propose we handle that error case?

Reply:
Ah okay we were talking about two different things here. For your point @blu3beri, I suggest we remove the metadata from the constructor and require it to be set from using the `record.metadata.set` afterwards.

What I was mainly thinking about here is the persistence. The metadata on the current credential records has a different structure than the metadata we're adding in this PR. My thinking was that we add something like this migration script called `0.1.0-alpha.291-0.1.0-alpha.292.tsx


```ts
export async function migrate291To292(agent: Agent) {
  const credentialRecords = await agent.credentials.getAll();
  const credentialRepository = await agent.injectionContainer.resolve(
    CredentialRepository
  );

  for (const credentialRecord of credentialRecords) {
    const data = credentialRecord.metadata.data;
    const metadata = MetaData();

    if (data.requestMetadata) {
      metadata.set("indyRequestMetadata", data.requestMetadata);
    }
    if (data.schemaId) {
      this.metadata.add("indyCredentialMetadata", {
        schemaId: data.schemaId,
      });
    }
    if (data.credentialDefinitionId) {
      this.metadata.add("indyCredentialMetadata", {
        credentialDefinitionId: data.credentialDefinitionId,
      });
    }

    credentialRecord.metadata = metadata;
    await credentialRepository.update(credentialRecord);
  }
}
```

Reply:
Ahh okay. I am in favour of the migration scripts, but for the 0.1.0 and so on. 👍

Reply:
Yeah I'd be in favor of migration scripts, as that'd be very nice for breaking changes. We could either tackle that now or at a 0.1.0 release, as you mentioned. :) 
I'll admit I'm not entirely sure how that should be handled from a mobile app perspective, but that can be figured out. :)